docs/reference/query-dsl/match-phrase-prefix-query.asciidoc

```diff
@@ -33,27 +25,74 @@ GET /_search
     "query": {
         "match_phrase_prefix" : {
             "message" : {
-                "query" : "quick brown f",
-                "max_expansions" : 10
+                "query" : "quick brown f"
             }
         }
     }
 }
 --------------------------------------------------
 // CONSOLE
 
-[IMPORTANT]
-===================================================
 
-The `match_phrase_prefix` query is a poor-man's autocomplete.  It is very easy
-to use, which lets you get started quickly with _search-as-you-type_ but its
-results, which usually are good enough,  can sometimes be confusing.
+[[match-phrase-prefix-top-level-params]]
+==== Top-level parameters for `match_phrase_prefix`
+`<field>`::
+(Required, object) Field you wish to search.
 
-Consider the query string `quick brown f`.  This query works by creating a
-phrase query out of `quick` and `brown` (i.e. the term `quick` must exist and
-must be followed by the term `brown`).  Then it looks at the sorted term
-dictionary to find the first 50 terms that begin with `f`, and
-adds these terms to the phrase query.
+[[match-phrase-prefix-field-params]]
+==== Parameters for `<field>`
+`query`::
++
+--
+(Required, string) Text you wish to find in the provided `<field>`. 
+
+The `match_phrase_prefix` query <<analysis,analyzes>> any provided text into
+tokens before performing a search. The last term of this text is treated as a
+<<query-dsl-prefix-query,prefix>>, matching any words that begin with that term.
+--
+
+`analyzer`::
+(Optional, string) <<analysis,Analyzer>> used to convert text in the `query`
+value into tokens. Defaults to the <<specify-index-time-analyzer,index-time
+analyzer>> mapped for the `<field>`. If no analyzer is mapped, the index's
+default analyzer is used.
+
+`max_expansions`::
+(Optional, integer) Maximum number of terms to which the last provided term of
+the `query` value will expand. Defaults to `50`.
+
+`slop`::
+(Optional, integer) Maximum number of positions allowed between matching tokens.
+Defaults to `0`. Transposed terms have a slop of `2`.
+
+`zero_terms_query`::
++
+--
+(Optional, string) Indicates whether no documents are returned if the `analyzer`
+removes all tokens, such as when using a `stop` filter. Valid values are:
+
+ `none` (Default)::
+No documents are returned if the `analyzer` removes all tokens.
+
+ `all`::
+Returns all documents, similar to a <<query-dsl-match-all-query,`match_all`>>
+query.
+--
+
+
+[[match-phrase-prefix-query-notes]]
+==== Notes
+
+[[match-phrase-prefix-autocomplete]]
+===== Avoid using the match phrase prefix query for search autocompletion
```

Comment:
I think "avoid" is a strong word here. I would rather say "Prefer not using".  Because for some use cases (for a quick start etc), it is a valid use case.

Reply:
I've changed this heading to "Using..."  with bb2e045. This will make the nuance of that use case a bit more clear and encourage users to read the block.